base/compiler/inferenceresult.jl

```diff
@@ -4,90 +4,89 @@ const EMPTY_VECTOR = Vector{Any}()
 
 mutable struct InferenceResult
     linfo::MethodInstance
-    args::Vector{Any}
-    vargs::Vector{Any} # Memoize vararg type info w/Consts here when calling get_argtypes
-                       # on the InferenceResult, so that the optimizer can use this info
-                       # later during inlining.
+    argtypes::Vector{Any}
     result # ::Type, or InferenceState if WIP
     src #::Union{CodeInfo, OptimizationState, Nothing} # if inferred copy is available
-    function InferenceResult(linfo::MethodInstance)
+    function InferenceResult(linfo::MethodInstance, caller_argtypes = nothing)
         if isdefined(linfo, :inferred_const)
             result = Const(linfo.inferred_const)
         else
             result = linfo.rettype
         end
-        return new(linfo, EMPTY_VECTOR, Any[], result, nothing)
+        return new(linfo, compute_inf_result_argtypes(linfo, caller_argtypes), result, nothing)
     end
 end
 
-function get_argtypes(result::InferenceResult)
-    result.args === EMPTY_VECTOR || return result.args # already cached
-    argtypes, vargs = get_argtypes(result.linfo)
-    result.args = argtypes
-    if vargs !== nothing
-        result.vargs = vargs
-    end
-    return argtypes
-end
-
-function get_argtypes(linfo::MethodInstance)
+function compute_inf_result_argtypes(linfo::MethodInstance, caller_argtypes = nothing)
     toplevel = !isa(linfo.def, Method)
-    atypes::SimpleVector = unwrap_unionall(linfo.specTypes).parameters
+    given_argtypes = Any[unwrap_unionall(linfo.specTypes).parameters...]
     nargs::Int = toplevel ? 0 : linfo.def.nargs
-    args = Vector{Any}(undef, nargs)
-    vargs = nothing
+    if !toplevel && caller_argtypes !== nothing
+        for i in 1:length(caller_argtypes)
+            a = maybe_widen_conditional(given_argtypes[i])
```

Comment:
Is this meant to be `caller_argtypes[i]`?

Reply:
Ah, yup! Thanks for that catch.